h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoderBuilder.java

```diff
@@ -0,0 +1,44 @@
+package ai.h2o.automl.targetencoding;
+
+import hex.ModelBuilder;
+import hex.ModelCategory;
+import water.DKV;
+import water.util.IcedHashMap;
+
+import java.util.Map;
+
+public class TargetEncoderBuilder extends ModelBuilder<TargetEncoderModel, TargetEncoderModel.TargetEncoderParameters, TargetEncoderModel.TargetEncoderOutput> {
```

Comment:
Interesting... I am not opposed to this idea but IMHO it would be better to do it in a different PR... this will make it much larger and harder to review.

Reply:
@michalkurka I tried but as I though before it is a problematic thing to do. TargetEncoderBuilder is highly coupled with TargetEncoderModel. TargetEncoderModel in turn is a dependency for TargetEncoderMojoWriter. This is how all this gets connected. I looked into your commit for Word2Vec and it is also contains both Model and Mojo classes in one PR. I could have separated TargetEncoderBuilder from TargetEncoderModel... but there is also dependency from model output to builder: 
```
public TargetEncoderOutput(TargetEncoderBuilder b) {
```
So I think that the best what I can do is to wait until two PRs(refactoring #3645, TargetEncoderBuilder with TargetEncoderModel #3646)  got reviewed and merged. Then I will rebase  Mojo Support PR #3282 onto master and it will be ready for a review with fewer changes. cc @tomkraljevic